### ISSUE TITLE
Graphic examples fail to quit

### ISSUE THREAD

**Issue by chimosky** [sugarlabs/Pippy#87]
Pressing `<ctrl>C` has no effect, killing the python process or `<ctrl>Q` does quit the activity.

Noticed this happens mostly with the graphics examples.

**Comment by chimosky** [comment]
@sparshg can you reproduce?

**Comment by sparshg** [comment]
I think this is working as intended, in graphics example with pygame, we have specified `ESCAPE` key in the event loop to exit the window that pygame spawns.

**Comment by chimosky** [comment]
> I think this is working as intended, in graphics example with pygame, we have specified `ESCAPE` key in the event loop to exit the window that pygame spawns.

Can you test and reproduce? I know that the escape keys are handled in the pygame events loop.

**Comment by sparshg** [comment]
I tested with pygame 2.4.0, and the escape key is working fine. I couldn't see any critical errors, can you mention the example in which you see the bug?

**Comment by chimosky** [comment]
Noticed it with all the graphics examples.

**Comment by quozl** [comment]
Pippy does mix pygame with GTK but does it differently to Sugargame or OLPCgame, somehow.  There may be some change in Sugargame that may need adopting in Pippy.  A quick look at activity.py shows a class PyGameActivity and the forking of a process to hide the pygame file inside.

**Comment by chimosky** [comment]
Agreed, if we decide to use sugargame here we might need to tweak it to suit Pippy.

**Comment by chimosky** [comment]
> Now, I wanted to clarify, is the issue that the ESCAPE key is not able to exit the pygame's full-screen window in graphics examples? Because in the description you mentioned <ctrl>c not being able to exit the pygame's window, which should be the intended behaviour. We can change this by including <ctrl>c in pygame's event loop to exit the pygame window.

Yes the issue is that `ESC` isn't able to exit the pygame full-screen window in graphics examples.
I mentioned the other key combinations as there's no other way to leave the full-screen window and exit the activity.

I'm testing with live build.

**Comment by quozl** [comment]
> I'm testing with live build.

Which one?

**Comment by chimosky** [comment]
Currently running 0.120.

**Comment by quozl** [comment]
Sorry, not familiar with that any more.  The version of Pygame and Python may be significant.

**Comment by sparshg** [comment]
Can you try running the example without pygame.NOFRAME flag we just introduced? Also which pygame and python version is it?

**Comment by chimosky** [comment]
> Can you try running the example without pygame.NOFRAME flag we just introduced? 

Doesn't make any difference.

> Also which pygame and python version is it?

pygame 2.3.0 and python3, tested on an F38 vm I have and it works fine.

I'll close this.
